libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -594,10 +594,14 @@ void NavEKF2_core::readGpsData()
             }
 
             // Read the GPS location in WGS-84 lat,long,height coordinates
-            const struct Location &gpsloc = gps.location();
+            Location gpsloc = gps.location();
 
             // Set the EKF origin and magnetic field declination if not previously set  and GPS checks have passed
             if (gpsGoodToAlign && !validOrigin) {
+                // if flying, correct for height change from takeoff so that the origin is at field elevation
+                if (inFlight) {
+                    gpsloc.alt += (int32_t)(100.0f * stateStruct.position.z);
```

Comment:
I think unfortunately you're going to need to create another Location object called something like "gpsloc_fieldelevation" and only adjust its altitude.  The issue is just that gpsloc is consumed a little lower down in the method as well.

Reply:
Fixed. Yup sorry about that one I should have checked better.